flink-connector-mongodb/src/main/java/org/apache/flink/connector/mongodb/source/enumerator/splitter/MongoSplitContext.java

```diff
@@ -0,0 +1,138 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.mongodb.source.enumerator.splitter;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.connector.mongodb.source.config.MongoReadOptions;
+
+import com.mongodb.MongoNamespace;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoCollection;
+import org.bson.BsonBoolean;
+import org.bson.BsonDocument;
+import org.bson.BsonInt64;
+
+import static org.apache.flink.connector.mongodb.common.utils.MongoConstants.AVG_OBJ_SIZE_FIELD;
+import static org.apache.flink.connector.mongodb.common.utils.MongoConstants.COUNT_FIELD;
+import static org.apache.flink.connector.mongodb.common.utils.MongoConstants.SHARDED_FIELD;
+import static org.apache.flink.connector.mongodb.common.utils.MongoConstants.SIZE_FIELD;
+
+/**
+ * The split context used by {@link MongoSplitters.MongoSplitter} to split collection into a set of
+ * chunks for MongoDB data source.
+ */
+@Internal
+public class MongoSplitContext {
+
+    /** Read options of MongoDB. */
+    private final MongoReadOptions readOptions;
+
+    /** Client of MongoDB. */
+    private final MongoClient mongoClient;
+
+    /** Namespace of MongoDB, eg. db.coll. */
+    private final MongoNamespace namespace;
+
+    /** Is a sharded collection. */
+    private final boolean sharded;
+
+    /** The number of objects or documents in this collection. */
+    private final long count;
+
+    /** The total uncompressed size(bytes) in memory of all records in a collection. */
+    private final long size;
+
+    /** The average size(bytes) of an object in the collection. */
+    private final long avgObjSize;
+
+    public MongoSplitContext(
+            MongoReadOptions readOptions,
+            MongoClient mongoClient,
+            MongoNamespace namespace,
+            boolean sharded,
+            long count,
+            long size,
+            long avgObjSize) {
+        this.readOptions = readOptions;
+        this.mongoClient = mongoClient;
+        this.namespace = namespace;
+        this.sharded = sharded;
+        this.count = count;
+        this.size = size;
+        this.avgObjSize = avgObjSize;
+    }
+
+    public static MongoSplitContext of(
+            MongoReadOptions readOptions,
+            MongoClient mongoClient,
+            MongoNamespace namespace,
+            BsonDocument collStats) {
+        return new MongoSplitContext(
+                readOptions,
+                mongoClient,
+                namespace,
+                collStats.getBoolean(SHARDED_FIELD, BsonBoolean.FALSE).getValue(),
+                collStats.getNumber(COUNT_FIELD, new BsonInt64(0)).longValue(),
+                collStats.getNumber(SIZE_FIELD, new BsonInt64(0)).longValue(),
+                collStats.getNumber(AVG_OBJ_SIZE_FIELD, new BsonInt64(0)).longValue());
```

Comment:
Also, 0 is a dangerous default for AVG_OBJ_SIZE_FIELD since it'd result in divide-by-zero in the sample partitioner.

Reply:
When the collection is empty, `avgObjSize` field will not be included in `collStats`. 
We should add some validation to avoid division by zero in the sample partitioner.

```json
{
    "ns": "test_source.empty_coll",
    "size": 0, 
    "count": 0, 
    "storageSize": 4096,
    "capped": false
}
```